operator/pkg/translate/translate.go

```diff
@@ -441,7 +441,8 @@ func (t *Translator) setComponentProperties(root map[string]interface{}, iop *v1
 		}
 
 		tag, found, _ := tpath.GetFromStructPath(iop, "Components."+string(cn)+".Tag")
-		if found && tag.(string) != "" {
+		tagStr, ok := tag.(string)
+		if found && !(ok && tagStr == "") {
```

Comment:
this will match if tag is not a string, is that what we want?

Reply:
string is the only case where we have to do something special because some overlay paths end up creating default "" when this is unset. i added a comment to clarify.